lib/cn/kvset.c

```diff
@@ -211,15 +212,53 @@ kvset_put_ref(struct kvset *ks)
     cn_work_submit(cn, kvset_put_ref_work, &ks->ks_kvset_cn_work);
 }
 
+static merr_t
+kvset_hblk_init(
+    struct mpool *mpool,
+    struct mblock_props *props,
+    struct mpool_mcache_map *map,
+    uint8_t **hlog,
+    struct kvset_hblk *blk)
+{
+    merr_t err;
+    struct kvs_mblk_desc *hbd = &blk->kh_hblk_desc;
+
+    err = hbr_read_desc(mpool, map, props, blk->kh_hblk.bk_blkid, hbd);
+    if (err)
+        return err;
+
+    err = hbr_read_seqno_range(hbd, &blk->kh_seqno_min, &blk->kh_seqno_max);
+    if (ev(err))
+        return err;
+
+    err = hbr_read_ptree_region_desc(hbd, &blk->kh_ptree_desc);
+    if (err)
+        return err;
+
+    err = hbr_read_metrics(hbd, &blk->kh_metrics);
+    if (err)
+        return err;
+
+    blk->kh_pfx_max = hbd->map_base + omf_hbh_max_pfx_off(hbd->map_base);
```

Comment:
In hblock_builder.c:make_header(), I see that you conditionally pack max_pfx and min_pfx into the header. Here, it is unconditionally unpacked. Please make sure it all falls in place when max_pfx and min_pfx are not packed.

Reply:
I am setting the offsets to unpacked, so this is fine.

Reply:
```
    if (max_pfx) {
        unsigned int max_pfx_len = 0;

        max_pfx_len = key_obj_len(max_pfx);
        assert(max_pfx_len <= HSE_KVS_PFX_LEN_MAX);

        omf_set_hbh_max_pfx_off(hdr, BYTE_ALIGN(HBLOCK_HDR_LEN - 2 * HSE_KVS_PFX_LEN_MAX));
        omf_set_hbh_max_pfx_len(hdr, max_pfx_len);

        key_obj_copy((void *)hdr + omf_hbh_max_pfx_off(hdr), HSE_KVS_PFX_LEN_MAX, NULL, max_pfx);
    }

    if (min_pfx) {
        unsigned int min_pfx_len = 0;

        min_pfx_len = key_obj_len(min_pfx);
        assert(min_pfx_len <= HSE_KVS_PFX_LEN_MAX);

        omf_set_hbh_min_pfx_off(hdr, BYTE_ALIGN(HBLOCK_HDR_LEN - HSE_KVS_PFX_LEN_MAX));
        omf_set_hbh_min_pfx_len(hdr, min_pfx_len);

        key_obj_copy((void *)hdr + omf_hbh_min_pfx_off(hdr), HSE_KVS_PFX_LEN_MAX, NULL, min_pfx);
    }
```

Reply:
Think I worded my comment poorly.

Let's say, max_pfx and min_pfx are both null in make_header(). Assuming the "hdr" buf in the pack routine is memset to zero, you would read back zeroes for pfx off, pfx len etc. here.. Just wanted to confirm that key_disc_init() handles this scenario gracefully.